distributed/nanny.py

```diff
@@ -921,7 +925,10 @@ def watch_stop_q(loop: IOLoop, worker: Worker) -> None:
             worker cleanly.
             """
             try:
-                msg = child_stop_q.get()
+                while True:
+                    msg = child_stop_q.get()
+                    if msg["op"] != "noop":
+                        break
```

Comment:
I don't get it; how can we have more than one noop in the queue?

Reply:
Also maybe we could rename it to "start" and put a comment which mentions that there's an explanation of the hack above?

Reply:
It's supposed to be ignored even if you were to send many `"noop"` messages

Reply:
But you're always going to have exactly one noop message, right?

Reply:
I've resolved this more cleanly by .get()-ing the message from the queue before passing it to AsyncProcess so the code in the child is unchanged